src/FieldWarning/Assets/Units/Component/Weapon/Cannon.cs

```diff
@@ -184,23 +184,7 @@ public sealed class Cannon
             }
 
             ShellBehaviour shellBehaviour = shell.GetComponent<ShellBehaviour>();
-            shellBehaviour.Initialize(shellDestination, ammo);
-
-            if (isServer)
-            {
-                if (target.IsUnit)
-                {
-                    if (isHit && !ammo.IsAoe)
-                    {
-                        target.Enemy.HandleHit(
-                                ammo.DamageType, ammo.DamageValue, displacement, distance);
-                    }
-                }
-                else
-                {
-                    // HE damage is applied by the shellBehavior when it explodes
-                }
-            }
+            shellBehaviour.Initialize(shellDestination, ammo, target.Enemy.gameObject.GetComponent<PFW.Units.Component.Movement.MovementComponent>().Velocity);
```

Comment:
Break up giga long lines if its not too much trouble.

Also normally this kind of thing would be done through the UnitDispatcher (add a UnitDispatcher property called 'CurrentMovementSpeed' or something and then have that get the value from movement component; the intent is to avoid non-unit code fiddling with specific components, and have the UnitDispatcher provide a clean interface for outside requests. Then this code just goes ```target.Enemy.CurrentMovementSpeed```. If it sounds like too much boilerplate you can leave it like this tho.